backend/src/v5/models/revisions.js

```diff
@@ -25,6 +26,27 @@ const excludeIncomplete = { incomplete: { $exists: false } };
 
 const collectionName = (model) => `${model}.history`;
 
+const findRevisionsByModel = async (teamspace, model, query, projection) => {
+	const revisions = await db.find(teamspace, collectionName(model), query, projection, {});
+
+	if (!revisions || revisions.length === 0) {
+		throw templates.containerNotFound;
+	}
+
+	return revisions;
+};
+
+const findOneRevisionByID = async (teamspace, model, revision, projection, sort) => {
+	const revisions = await findRevisionsByModel(teamspace, model, {}, projection, sort);
```

Comment:
You should just use `findOneRevisionByQuery`  where `_id: revision`. If you are looking for a single entry you should never draw all the data out and filter here - let the database do the heavy lifting.

revision id should have been converted prior to the data entering the processor - so you shouldn't be doing any UUID to string converasions here.. This was explained on the blurb i sent out before I went on holiday: 
![image](https://user-images.githubusercontent.com/11945337/133469721-8eb561ae-66aa-4af9-9327-de62a60d93c1.png)

Reply:
Regarding the first one I did that just to have an extra check that the revisions table exists cause I thought that by the time you create a model we automatically generate a revisions history table

Reply:
no we don't , nor does it matter